### ISSUE TITLE
Compiler crash when accessing constant variable before declaration

### ISSUE THREAD

**Issue by geominorai** [alliedmodders/sourcepawn#673]
# Help us help you
  - [x] I have checked that my issue [doesn't exist yet](https://github.com/alliedmodders/sourcemod/issues).
  - [x] I have tried my absolute best to reduce the problem-space and have provided the absolute smallest test-case possible.
  - [x] I can always reproduce the issue with the provided description below.

# Environment
  * Current SourceMod snapshot: `1.11.0.6745`

# Description
Compiler crashes when a `const` variable is used before declaration on the 1.11 compiler but prints errors on 1.10 (although some are strange).

# Problematic Code
```cpp
public void OnPluginStart() {
	int iVal = iConst;
	const int iConst = 5;
}
```
# Expected Behavior
```
SourcePawn Compiler 1.10.0.6502
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2018 AlliedModders LLC

C:\Build\SM1.10\scripting\spcomp_const_bug.sp(2) : error 017: undefined symbol "iConst"
C:\Build\SM1.10\scripting\spcomp_const_bug.sp(2) : error 130: cannot coerce functions to values
C:\Build\SM1.10\scripting\spcomp_const_bug.sp(3) : error 021: symbol already defined: "iConst"

3 Errors.
```

# Actual Behavior
```
SourcePawn Compiler 1.11.0.6745
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2018 AlliedModders LLC
```
Returns exit code `0xc0000005` (Access Violation)

**Comment by asherkin** [comment]
This behaves correctly on master, so it probably isn't worth digging in to fix it given the code is obviously wrong.

```
SourcePawn Compiler 1.12
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2021 AlliedModders LLC

bug673.sp(2) : error 017: undefined symbol "iConst"

1 Error.
```

**Comment by geominorai** [comment]
Glad to hear it is fixed there.  It just took me a bit to figure out why moving some lines of code around in my project suddenly triggered no output from the compiler.

**Comment by dvander** [comment]
My current plan is to abandon the 1.11-dev tag for SourcePawn, and to have SourceMod track the master branch again. This will ultimately resolve the problem. This change will hopefully happen in the next few weeks.

**Comment by geominorai** [comment]
Would it be bundled would the SM 1.11 releases though?

**Comment by dvander** [comment]
Yes, SM will start picking changesets off the master branch again. Eg SP master will become 1.11 instead of 1.12.
